#12312: panoramax streetview support

nrc97 opened the pull request:
## Description
This PR introduces support for Panoramax as an additional Street View provider in MapStore2. Users can now visualize street‑level imagery using Panoramax alongside existing providers. The feature includes configuration options, i18n strings, attribution handling, and provider selection at runtime.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
#12312 
Panoramax is not supportted as streetview provider.

**What is the new behavior?**
We can now use panoramax as provider for the streetview plugin.
<img width="2290" height="1261" alt="Capture d&#39;écran 2025-12-16 122255" src="https://github.com/user-attachments/assets/e1598b92-52a3-408f-bb36-052646608e56" />

## Breaking change
 - [x] No

## Other useful information
Panoramax OpenAPI Documentation: https://docs.panoramax.fr/api/api/openapi/
Panoramax Viewer Documentation: https://docs.panoramax.fr/web-viewer/#basic-usage

nrc97 committed:
Adapt Mapstore2 for Panoramax support

map.forEachFeatureAtPixel in openlayers/Map.jsx adapted for detection of features of type RenderFeature used for MVT layers. By doing that, MVT layers are clickable and the intersected features are return on clic
TileProviderLayer for being able to create MVT layer and display it.

nrc97 committed:
Add and configure Panoramax plugin in StreetView component

- Create a viewer using pnx-photo-viewer to display pictures.
- Add a Panoramax VectorTileLayer using OpenLayers styling to match other providers.
- Handle click events: retrieve picture data directly from the layer datas, or call the API if the data is not found.

nrc97 committed:
Translate panoramax emptySelection message in French, English, Spanish, Italian and German

Gaetanbrl commented:
Is this a tool dedicated to OpenLayers ? (Mapstore offering several map libraries)

tdipisa reviewed (changes requested):
@nrc97 conflicts fix needed.

tdipisa commented:
> In order to contribute to the MapStore project, the CLA (Contributor License agreement) should be sent signed to GeoSolutions. Please consult contributing rules at: https://github.com/geosolutions-it/MapStore2/wiki/Contributing-to-MapStore#contributing-code

@nrc97 please notice that, thank you.

nrc97 committed:
Remove unused DEFAULT_SRS constant

Removed unused DEFAULT_SRS constant from panoramax.js to match the eslint configuration

FNI18300 commented:
Hello @landryb ,
all CLAs have been signed for OPEN contributors and sent to @tdipisa on Wednesday, March 25, 2026.

This includes me and OPGuillaume, Mbayyari, Fhunkeler and of course Nrc97

sigrmed commented:
Hello @tdipisa, could you please review this PR, we'd like to test this functionnality. Thank you so much.

landryb commented:
> Hello @landryb , all CLAs have been signed for OPEN contributors and sent to @tdipisa on Wednesday, March 25, 2026.
> 
> This includes me and OPGuillaume, Mbayyari, Fhunkeler and of course Nrc97

thanks, that's great news ! :)

tdipisa commented:
@sigrmed @nrc97 the review will be done today. Thank you for your patience.

offtherailz commented on web/client/translations/data.fr-FR.json:
```diff
@@ -4402,7 +4402,9 @@
           "projectionNotAvailable": "La projection {srs} configurée n'est pas disponible. Veuillez contacter l'administrateur."
         },
-        "reloadAPI": "Recharger l'API"
+        "reloadAPI": "Recharger l'API"},
```
```suggestion
        "reloadAPI": "Recharger l'API"
      },
```

offtherailz commented on web/client/plugins/StreetView/StreetView.jsx:
```diff
@@ -115,6 +115,23 @@ const StreetViewPluginContainer = connect(() => ({}), {
+ *         "provider": "panoramax",
+ *         "providerSettings": {
+ *           "srs": "EPSG:4326",
+ *           "PanoramaxApiURL": "https://api.panoramax.xyz/api"
```
Let's commit a valid documentation
```suggestion
 *           "PanoramaxApiURL": "https://api.panoramax.xyz/api",
```

offtherailz commented on web/client/components/map/openlayers/Map.jsx:
```diff
@@ -386,6 +389,30 @@ class OpenlayersMap extends React.Component {
+        case 'MultiPolygon': return new MultiPolygon([coords]);
+        default: return null; // types not supported
+        }
+    };
```
This error do not allows the build at all
```suggestion
    };
    /**
```

offtherailz committed:
minor fixes

offtherailz commented:
Hi,
I'm going to review this PR today.

offtherailz commented on web/client/plugins/StreetView/selectors/streetView.js:
```diff
@@ -120,6 +129,22 @@ const CYCLOMEDIA_DATA_LAYER_DEFAULTS = {
 };
+
+const PANORAMAX_DATA_LAYER_DEFAULTS = {
+    type: 'tileprovider',       // utilise le plugin TileProvider OpenLayers
+    provider: 'custom',         // “custom” pour donner directement une URL
+    url: 'https://panoramax.openstreetmap.fr/api/map/{z}/{x}/{y}.mvt',
+    format: 'application/vnd.mapbox-vector-tile', // signale que c’est du MVT
+    name: 'panoramax:sequences',
+    visibility: true,
+    // optionnel: bornes de zoom
```
Always use english

```suggestion
    type: 'tileprovider',       // use The plugin TileProvider for OpenLayers
    provider: 'custom',         // “custom” provider to use the URL
    url: 'https://panoramax.openstreetmap.fr/api/map/{z}/{x}/{y}.mvt',
    format: 'application/vnd.mapbox-vector-tile', // signale que c’est du MVT
    name: 'panoramax:sequences',
    visibility: true,
    // optional: zoom levels
```

offtherailz replied:
Use english

offtherailz committed:
Fix style support for RenderFeature

offtherailz committed:
fixed more french issues

offtherailz committed:
fixed json style

offtherailz commented on web/client/plugins/StreetView/selectors/streetView.js:
```diff
@@ -95,6 +96,14 @@ const getVectorStyle = (overrides) => ({
     }
 });
+
+const getPanoramaxMVTStyle = () => {
+    // For a better integration we use the same radius, width, color and opacity as the geostyler style
+    return new Style({
+        stroke: new Stroke({ color: '#3165EFCC', width: 2}),
+        image: new CircleStyle({ radius: 6, fill: new Fill({ color: '#3165EF99' }) })
+    });
+};
```
This will put in redux the OL style, we should use the current vector styling system instead.

offtherailz reviewed (changes requested):
Hi,
Thank you for contributing. :+1:  You may see my inline suggestions **but read the text below first** I did most of the fixes.

Here my notes
1. I found some issues you may have committed in error.
2. The functionality, once fixed the issues was OK. Anyway there were 2 points to take into account:
    1. The style applied this way is problematic. 
    2. The usage of tileprovider can be correct, but we need to implement vector tiles, so this is a partial support for TMS. 

For 1 and 2.1 I did [this PR to your branch ](https://github.com/groupe-open-rennes/MapStore2/pull/1/changes) with the fixes. Once merged, the PRs should work properly. 

For point 2.2 it's ok. Take into account that we may change this implementation in the future, with a more organic support to vector tiles. 

So long story short, [if you apply changes in this PR](https://github.com/groupe-open-rennes/MapStore2/pull/1/changes) it should be ok. Please merge my changes, align to master, test everything once more and send me the review again.

nrc97 commented:
Hello @offtherailz 

I merged your PR and tested it.
I had to make the following changes:
- Translation in english of one last remaining french comment
- The line rule was named "Circle" as the circle rule in the geostyler style for the panoramax layer, I changed it to "Line"

The PR has also been rebased on the master branch and here is a screenshot of the test I made
<img width="3434" height="1194" alt="image" src="https://github.com/user-attachments/assets/09e3e3e1-9307-4bc6-8055-b12f1fc84505" />

tdipisa commented:
@nrc97 I'm sorry but it seems there is some issue with the CLA recognition which is no longer passing the automated check. It was set to Nrc97 account, and the check was passing successfully, while now I see it is nrc97. Since you already have signed the CLA form, I've updated the .clabot file accordingly. To pass this check you need to align your PR to the latest master, please.

nrc97 commented:
> @nrc97 I'm sorry but it seems there is some issue with the CLA recognition which is no longer passing the automated check. It was set to Nrc97 account, and the check was passing successfully, while now I see it is nrc97. Since you already have signed the CLA form, I've updated the .clabot file accordingly. To pass this check you need to align your PR to the latest master, please.

Thank you @tdipisa
It was a mistake, it should be nrc97

tdipisa commented:
> > @nrc97 I'm sorry but it seems there is some issue with the CLA recognition which is no longer passing the automated check. It was set to Nrc97 account, and the check was passing successfully, while now I see it is nrc97. Since you already have signed the CLA form, I've updated the .clabot file accordingly. To pass this check you need to align your PR to the latest master, please.
> 
> Thank you @tdipisa It was a mistake, it should be nrc97

ok @nrc97 thank you for clarifying that. Anyway you have to align the branch you have used for this PR. Many thanks.

nrc97 committed:
Fix typo and french issues

nrc97 committed:
return pre computed geom instead of getting it from the original feature in startpoint/endpoint geom function

tdipisa commented:
@nrc97 All good now regarding the CLA. I just realized there is no issue associated to this PR. It should be, please. See also the contributing rules [here](https://github.com/geosolutions-it/MapStore2/wiki/Contributing-to-MapStore). Can you create it please and include the link in the PR description? Thanks in advance for this last step.

offtherailz commented:
I created the #12312 issue for this, in order to move on and finalize this task.
